tests/ratingservice/main_test.py

```diff
@@ -18,76 +18,126 @@
 import sys
 import unittest
 import requests
-from requests.adapters import HTTPAdapter
+from requests.exceptions import RequestException
+from urllib3.exceptions import ReadTimeoutError
 import json
 import decimal
 
 
-service_url = ''
-products = []
-
-
-def composeUrl(resource, eid=""):
-    if not eid:
-        return "{0}/{1}".format(service_url, resource)
-    else:
-        return "{0}/{1}/{2}".format(service_url, resource, eid)
-
-
 class TestEndpoints(unittest.TestCase):
 
+    @classmethod
+    def composeUrl(cls, resource, eid=""):
+        if not eid:
+            return "{0}/{1}".format(cls.service_url, resource)
+        else:
+            return "{0}/{1}/{2}".format(cls.service_url, resource, eid)
+
+    @classmethod
+    def setUpClass(cls):
+        cls.TIMEOUT = 5.0
+
+        cls.service_url = ''
+        if len(sys.argv) > 1:
+            cls.service_url = sys.argv[1].rstrip('/')
+
+        cls.products = []
+        path = os.getcwd().rstrip('/') + '/src/productcatalogservice/products.json'
```

Comment:
nit: comments would be useful here. You may even want to keep the `getProducts`, whatever you think is most readable

Reply:
I will add commenting 👍